signer/threshold_validator.go

```diff
@@ -42,36 +44,34 @@ type ThresholdValidator struct {
 	logger log.Logger
 }
 
-type ThresholdValidatorOpt struct {
-	Pubkey    crypto.PubKey
-	Threshold int
-	SignState SignState
-	Cosigner  Cosigner
-	Peers     []Cosigner
-	RaftStore *RaftStore
-	Logger    log.Logger
-}
-
 // NewThresholdValidator creates and returns a new ThresholdValidator
-func NewThresholdValidator(opt *ThresholdValidatorOpt) *ThresholdValidator {
-	validator := &ThresholdValidator{}
-	validator.cosigner = opt.Cosigner
-	validator.peers = opt.Peers
-	validator.threshold = opt.Threshold
-	validator.pubkey = opt.Pubkey
-	validator.lastSignState = opt.SignState
-	validator.lastSignStateMutex = sync.Mutex{}
-	validator.lastSignStateInitiated = SignState{
-		Height:   opt.SignState.Height,
-		Round:    opt.SignState.Round,
-		Step:     opt.SignState.Step,
-		filePath: "none",
-		cache:    make(map[HRSKey]SignStateConsensus),
+func NewThresholdValidator(
+	config *RuntimeConfig,
+	pubKey crypto.PubKey,
+	signState SignState,
+	threshold int,
+	cosigner Cosigner,
+	peers []Cosigner,
+	raftStore *RaftStore,
+	logger log.Logger,
```

Comment:
nit: I'll pretend to be Mark and say the logger should be the first argument to this function. 😆

Reply:
Eight positional arguments is a bit much here too.

Putting the logger first (or second if the function accepts `context.Context`) is a simple convention that takes guesswork out of where you should place it. Putting it last can't work with variadic functions.